ssh-encoding/src/error.rs

```diff
@@ -35,6 +35,9 @@ pub enum Error {
         /// Number of bytes of remaining data at end of message.
         remaining: usize,
     },
+
+    /// Invalid discriminant value in message.
+    InvalidDiscriminant(u128),
```

Comment:
It seems weird to me to conflate custom derive errors with this error type. Perhaps you could just define an error type in `ssh-derive`?